### DISCUSSION TITLE
Custom document_class not being returned

### DISCUSSION THREAD

**Discussion by markbrai** [msiemens/tinydb#476] [Q&A] [answered]
Hi,

I am trying to define a custom document class for my use to provide some 'dot' access to the main fields of the database.

For example, if I have the field 'name' in my database, I would specify the custom class such as this:

```python
class MyDocument(Document):
    def __init__(self, value, doc_id):
        super().__init__(value, doc_id)
        self.name = self["name"]
```

Then, in my script I would import TinyDB and assign `MyDocument` to `TinyDB.document_class`

```python
from tinydb import TinyDB, Query

TinyDB.document_class = MyDocument

db = TinyDB("db.json")

db.insert({"name":"John Smith"})
```

Then, running `type(db.get())` like this:

```python
Name = Query()

type(db.get(Name.name == "John Smith"))

>>> tinydb.table.Document
```

Whereas I would expect it should be `MyDocument`. If I run `db.document_class` though, I get:

```python
db.document_class

>>> __main__.MyDocument
```

So it looks like it is assigning correctly, but the return is still as a Document?

Any hints appreciated! It may well be that I am just using this for completely the wrong thing...

Cheers,

**Comment by msiemens** [comment, marked as answer]
Hey @markbrai, I think I understand this issue: You're adding the `document_class` attribute to the `TinyDB` class instead of assigning it to the `Table` class. You can check this by trying to see the `document_class` value before trying to change it:

```
>>> print(TinyDB.document_class)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: type object 'TinyDB' has no attribute 'document_class'
```

As you can see, the `TinyDB` class has no attribute called `document_class`. But when you try to assign it, Python will not throw an error but instead create a new attribute for this class. Instead, try setting the property like this:

```
TinyDB.table_class.document_class = MyDocument
```

Then it should work 🙂

**Comment by markbrai** [comment]
Perfect! I knew I was missing something basic - I should have noticed this as I was having to import Document from tinydb.table to start with!

Working perfectly now - thankyou!
